plugins/plugin-core-support/src/lib/new-tab.ts

```diff
@@ -61,27 +62,50 @@ const getTabCloser = (tab: Tab) => getTabButton(tab).querySelector('.left-tab-st
  */
 class TabState {
   /** environment variables */
-  readonly env: Record<string, string>
+  private _env: Record<string, string>
 
   /** current working directory */
-  readonly cwd: string
+  private _cwd: string
 
-  constructor() {
-    this.env = Object.assign({}, process.env)
-    this.cwd = process.cwd().slice(0) // just in case, copy the string
```

Comment:
The state of the current tab is intialized and captured when **another** tab is opened. With this behavior, we can't store the watching jobs without creating global variable. Thus, I changed to initialize the current tab with the state being initialized as well, and we could add the watching job to the current tab.